src/Core/Utilities/CoreHelpers.cs

```diff
@@ -76,6 +76,39 @@ internal static Guid GenerateComb(Guid startingGuid, DateTime time)
         return new Guid(guidArray);
     }
 
+    internal static DateTime DateFromComb(Guid combGuid)
+    {
+        var guidArray = combGuid.ToByteArray();
+        var daysArray = new byte[4];
+        var msecsArray = new byte[4];
+
+        Array.Copy(guidArray, guidArray.Length - 6, daysArray, 2, 2);
+        Array.Copy(guidArray, guidArray.Length - 4, msecsArray, 0, 4);
+
+        Array.Reverse(daysArray);
+        Array.Reverse(msecsArray);
+
+        var days = BitConverter.ToInt32(daysArray, 0);
+        var msecs = BitConverter.ToInt32(msecsArray, 0);
+
+        var time = new TimeSpan(days, 0, 0, 0, (int)(msecs * 3.333333));
+        return new DateTime(_baseDateTicks + time.Ticks, DateTimeKind.Utc);
+    }
+
+    internal static long BinForComb(Guid combGuid, int binCount)
+    {
+        // From System.Web.Util.HashCodeCombiner
+        uint CombineHashCodes(uint h1, byte h2)
+        {
+            return (uint)(((h1 << 5) + h1) ^ h2);
+        }
+        var guidArray = combGuid.ToByteArray();
+        var randomArray = new byte[10];
+        Array.Copy(guidArray, 0, randomArray, 0, 10);
+        var hash = randomArray.Aggregate((uint)randomArray.Length, CombineHashCodes);
+        return hash % binCount;
```

Comment:
Array hash codes, even the `IStructuralEquatable` implementation, seems to include object information from the Array.

This hash method is extracted from the [Array implementation of `GetHashCode` for `IStructuralEquatable`](https://github.com/mono/mono/blob/main/mcs/class/referencesource/mscorlib/system/array.cs#L807-L810). I included the `uint` length as a seed to avoid degeneracy of different length Arrays filled exclusively with `0`, unlikely as that is.

Reply:
Unfortunately `HashCode` uses a global seed on application startup, so we have to stick with what we have